Assets/Scripts/PlayerController.cs

```diff
@@ -0,0 +1,227 @@
+            Flip();
+        }
+        else if (moveHorizontal < 0 && isFacingRight)
+        {
+            Flip();
+        }
+    }
+
+    private void Jump()
+    {
+        anim.SetFloat("playerVSpeed", rb2d.velocity.y);
+
+        if (Input.GetButtonDown("Jump") && grounded && !isDead)
+        {
+            rb2d.AddForce(Vector2.up * jumpForceUp);
+
+            if (isFacingRight)
+            {
+                rb2d.AddForce(Vector2.right * jumpForceForward);
+            }
+            else if (!isFacingRight)
+            {
+                rb2d.AddForce(Vector2.left * jumpForceForward);
+            }
+        }
+    }
+
+    private void Shoot()
+    {        
+        if (Input.GetButtonDown("Fire1") && overheat < 100)
+        {
+            anim.SetBool("isShooting", true);
+            overheat += overheatAdd;
+            rb2d.velocity = Vector3.zero;
+            anim.SetTrigger("playerShoot");
+            Vector2 recoil = new Vector2();
+            recoil = transform.TransformDirection(-firePoint.right * recoilSpeed);
+
+            Rigidbody2D bulletClone;
+
+            bulletClone = Instantiate(bullet, transform.position + firePoint.right, firePoint.transform.rotation) as Rigidbody2D;
+            bulletClone.velocity = transform.TransformDirection(firePoint.right * bulletSpeed);
+
+            rb2d.AddForce(recoil, ForceMode2D.Impulse);
+
+            if (isFacingRight && (firePoint.rotation.eulerAngles.z > 90 && firePoint.rotation.eulerAngles.z < 270))
+            {
+                Flip();
+            }
+            else if (!isFacingRight && (firePoint.rotation.eulerAngles.z < 90 || firePoint.rotation.eulerAngles.z > 270))
+            {
+                Flip();
+            }
+        }
+    }
+
+    private void GroundCheck()
+    {
+        grounded = groundDetectTrigger.OverlapCollider(groundContactFilter, groundHitDetectionResults) > 0;
+        anim.SetBool("isGrounded", grounded);
+    }
+
+    private void Flip()
+    {
+        isFacingRight = !isFacingRight;
+        Vector3 theScale = transform.localScale;
+        theScale.x *= -1;
+        transform.localScale = theScale;
+    }
+
+    private void SetCurrentCheckpoint(Checkpoint newCurrentCheckpoint)
+    {
+        if (currentCheckpoint != null)
+        {
+            currentCheckpoint.SetIsActivated(false);
+        }
+
+        currentCheckpoint = newCurrentCheckpoint;
+        currentCheckpoint.SetIsActivated(true);
+    }
+
+    private void Respawn()
+    {
+        rb2d.velocity = Vector2.zero;
+        anim.ResetTrigger("playerShoot");
+        grounded = true;
+        overheat = 0;
+
+        if (currentCheckpoint == null)
+        {
+            SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+        }
+        else
+        {
+            rb2d.velocity = Vector2.zero;
+            transform.position = currentCheckpoint.transform.position;
+        }
+        isDead = false;
+    }
+
+    private void UpdateCollectables()
```

Comment:
I don't like this name, I'd name it more like "UpdateCollectableText" since you're updating the text, not the collectibles themselves.